### PR TITLE
Revise error message for deprecated(old) style of k definitions.

### PR DESCRIPTION
**Pull request by xc93**
Revising the error message when parsing a K definition that uses old fashion of ' \` ' instead of parenthesis. The new error message looks the following.

[Error] Compiler: Had 2 parsing errors.
[Error] Inner Parser: Parse error: unexpected character '*'.Please use '(' or
')' instead of '\`'
        Source(./test.k)
        Location(10,29,10,30)
[Error] Inner Parser: Parse error: unexpected character '*'.Please use '(' or
')' instead of '\`'
        Source(./test.k)
        Location(11,24,11,25)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/148313234996210/224288625852417)

### CODE REVIEW

File: kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java [hunk 1]
```diff
@@ -574,9 +574,11 @@ public Term parse(NonTerminal nt, int position) {
             CharSequence content = s.input;
             ParseError perror = getErrors();
 
+            //Here we fix the error message for bug #1948
             String msg = content.length() == perror.position ?
                     "Parse error: unexpected end of file." :
                     "Parse error: unexpected character '" + content.charAt(perror.position) + "'.";
+            if (content.toString().indexOf('`')>=0) msg += "Please use '(' or ')' instead of '`'";
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java]: This can actually fail badly, and the error message can be misleading in some cases.
If the language actually includes a terminal backtick, and the user has a proper error at that location, now K would suggest using parenthesis instead of actually fixing the problem.

**Comment by radumereuta** [comment]
The backtick modifications started after this issue was reported: #1970 and #2028.
I think for #1948 we should just manually fix the backticks and continue debugging from there.

Also, when doing pull request, try to keep the original alignment, because it makes code reviews a lot harder.

**Comment by xc93** [comment]
@radumereuta I changed backtick to parenthesis and everything works fine under the latest version of K. So what is the real problem here?

**Comment by cos** [comment]
I agree with @radumereuta. The backticks haven't been in use long enough to warrant a special error message. The alternate "fix" for this is to go through the existing definitions and check whether there are any leftover backticks. If there are none -- likely -- great. If there are some, we should simply replace them with parenthesis.

File: kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java [hunk 2]
```diff
@@ -668,22 +670,22 @@ private boolean finishStateReturn(StateReturn stateReturn) {
...
-                    stateReturn.key.stateEnd)).function);
+                    stateReturn.key.stateCall.function,
+                    s.stateReturns.get(new StateReturn.Key(
+                            s.stateCalls.get(new StateCall.Key(
+                                    s.ntCalls.get(new NonTerminalCall.Key(
+                                            ((Grammar.NonTerminalState) stateReturn.key.stateCall.key.state).child,
+                                            stateReturn.key.stateCall.key.stateBegin)),
+                                    stateReturn.key.stateEnd,
+                                    ((Grammar.NonTerminalState) stateReturn.key.stateCall.key.state).child.exitState)),
+                            stateReturn.key.stateEnd)).function);
```
Comment (cos) [inline kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java:688]: please do not reformat the files.

File: kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java (hunk 1 above)
Comment (cos) [reply kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Parser.java]: maybe rephrase as: "Using backtick, '`', for K brackets is deprecated. If you are using it for this purpose, please use regular parenthesis, '(' and ')', instead."

**Review by cos — CHANGES_REQUESTED** [review]

**Comment by daejunpark** [comment]
Close this PR exercise in the class.
